### ISSUE TITLE
Inactivity timeout ignored on computer sleep

### ISSUE THREAD

**Issue by samuraiii** [keepassxreboot/keepassxc#2014]
I have inactivity timeout set on keepasxc, but database is not locked after longer time in sleep.
## Expected Behavior

Database should lock after set inactivity time even if computer was sleeping during that time  (at least it should lock itself after waking.

## Current Behavior

Database is not locked even after 12 hours of sleep.

## Steps to Reproduce (for bugs)

1. install keepasxc
2.create db and open it
3.set inactivity timeout for lock
4.put computer to sleep and wait for timeout to pass
5.wake computer and find unlocked db

## Debug Info

KeePassXC - Version 2.2.4
Revision: 4723f66

Libraries:
- Qt 5.9.4
- libgcrypt 1.8.1

Operating system: Gentoo/Linux
CPU architecture: x86_64
Kernel: linux 4.15.18-gentoo

Enabled extensions:
- KeePassHTTP
- Auto-Type

**Comment by droidmonkey** [comment]
This may be a bug, but why not activate lock db on screen lock?

**Comment by orzechow** [comment]
I experience this issue nearly every day!

At work I need my passwords multiple times per hour, while I lock my screen very frequently (for meetings, nature’s call, …). 
I prefer to have to unlock the database max. 2-3 times per day.
So automatic database locking on screen lock is not an option for me.

Thus I use the inactivity timeout to lock the database.

But as I suspend the session at the end of a workday (Ubuntu based system, suspend to RAM), the database is often still unlocked on the next morning.
I would expect KeepassXC to at least lock the database after system wakeup.

An even safer solution would be to lock the database when suspending right away (without locking on session lock). As this is a little different feature, I've created a new issue for that: #2753

**Comment by Spaak** [comment]
> I experience this issue nearly every day!
> 
> At work I need my passwords multiple times per hour, while I lock my screen very frequently (for meetings, nature’s call, …). I prefer to have to unlock the database max. 2-3 times per day. So automatic database locking on screen lock is not an option for me.
> 
> Thus I use the inactivity timeout to lock the database.
> 
> But as I suspend the session at the end of a workday (Ubuntu based system, suspend to RAM), the database is often still unlocked on the next morning. I would expect KeepassXC to at least lock the database after system wakeup.

Yep, exactly the same situation for me, so inactivity timeout based on wall clock would be very helpful.

> An even safer solution would be to lock the database when suspending right away (without locking on session lock). As this is a little different feature, I've created a new issue for that: #2753

Just chiming in to say that this suggestion #2753 would *not* be a solution on MacOS, since suspend/locking is managed quite differently there (you never actually "suspend" the system, just lock with the button or close the lid and MacOS figures out what to do in terms of power management--open lid always immediately wakes up no matter the downtime in between). The above (use walltime for inactivity locking timer) *would* be a solution on MacOS.

**Comment by buffytepper** [comment]
I'm having this issue lately. When I first set it up, it had no issue. I set mine to lock after a business day and I have the same reasons as @orzechow. Whilst it's not working right now, I will choose to lock the db when my session/screen is locked. But it's not ideal as I will have to enter this password at a minimum several times per day.

edit: I now see my system (company managed) has the power settings to turn off after a given period of inactivity. So, it now makes sense why this happens all of a sudden. Still, I'm hoping that there is a fix in development
